fuse_core/include/fuse_core/util.h

```diff
@@ -178,6 +181,45 @@ T getPositiveParam(const ros::NodeHandle& node_handle, const std::string& parame
   return value;
 }
 
+/**
+ * @brief Helper function that loads a covariance matrix diagonal vector from the parameter server and checks the size
+ * and the values are invalid, i.e. they are positive.
+ *
+ * @tparam Scalar - A scalar type, defaults to double
+ * @tparam Size - An int size that specifies the expected size of the covariance matrix (rows and columns)
+ *
+ * @param[in] node_handle - The node handle used to load the parameter
+ * @param[in] parameter_name - The parameter name to load
+ * @param[in] default_value - A default value to use for all the diagonal elements if the provided parameter name does
+ *                            not exist
+ * @return The loaded (or default) covariance matrix, generated from the diagonal vector
+ */
+template <int Size, typename Scalar = double>
+fuse_core::Matrix<Scalar, Size, Size> getCovarianceMatrixDiagonalParam(const ros::NodeHandle& node_handle,
```

Comment:
I'm open to suggestions to change this function name. Maybe `getCovarianceDiagonalParam` is enough, so at least it's a little smaller.

I'm also not sure if this is the best helper function we could provide for this, or whether it should be divided in some additional parts.

Reply:
Either way. I assume if you've noticed anything about my coding style, it's that I don't mind long names.

Reply:
Renamed to `getCovarianceDiagonalParm`, since covariance implies matrix.